app/lib/features/chat/widgets/receipts_builder.dart

```diff
@@ -1,18 +1,114 @@
 import 'package:acter/common/providers/chat_providers.dart';
 import 'package:acter/common/themes/app_theme.dart';
 import 'package:acter_avatar/acter_avatar.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart'
+    show Convo, EventSendState;
+import 'package:atlas_icons/atlas_icons.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:quds_popup_menu/quds_popup_menu.dart';
+import 'package:flutter_chat_types/flutter_chat_types.dart' as types;
 
 class ReceiptsBuilder extends ConsumerWidget {
-  final List<String> seenList;
-
+  final Convo convo;
+  final types.Message message;
   const ReceiptsBuilder({
     super.key,
-    required this.seenList,
+    required this.convo,
+    required this.message,
   });
 
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final receipts = message.metadata?['receipts'];
+    EventSendState? sendState = message.metadata?['eventState'];
+    if (receipts != null && receipts.isNotEmpty) {
+      return _UserReceiptsWidget(
+        seenList: (receipts as Map<String, int>).keys.toList(),
+      );
+    } else {
+      if (sendState != null) {
+        switch (sendState.state()) {
+          case 'NotSentYet':
+            return const SizedBox(
+              height: 8,
+              width: 8,
+              child: CircularProgressIndicator(),
+            );
+          case 'SendingFailed':
+            return Row(
+              children: <Widget>[
+                GestureDetector(
+                  onTap: () => _handleCancelRetrySend(),
+                  child: Text(
+                    'Cancel Send',
+                    style: Theme.of(context).textTheme.labelSmall!.copyWith(
+                          color: Theme.of(context).colorScheme.neutral5,
+                          decoration: TextDecoration.underline,
+                        ),
+                  ),
+                ),
+                const SizedBox(
+                  width: 10,
+                ),
+                GestureDetector(
+                  onTap: () => _handleRetry(),
+                  child: RichText(
+                    text: TextSpan(
+                      text: 'Failed to sent: ${sendState.error()}. ',
```

Comment:
[B](text: 'Failed to sent: ${sendState.error()}.)
```suggestion
                      text: 'Failed to sent.',
```

Better to now show technical errors to end users.

Reply:
Disagreed. As it makes a huge difference to the user whether the message wasn't sent because there is no internet vs. they are not authorized to sent it. As we don't have proper types for the errors and thus can't show a more nuanced message, at least showing the actual error (even just so we can see it in a screenshot they report), is better than nothing.